lib/grape/app/initializers/pre.rb

```diff
@@ -43,6 +44,15 @@ def call(exception, *)
     ActiveRecord::Base.default_timezone = :utc
   end
 
+  # legacy_connection_handling was deprecated in ActiveRecord 7
+  if ActiveRecord::VERSION::MAJOR < 7
```

Comment:
why not just `elsif ActiveRecord::Base.respond_to?(:legacy_connection_handling=)`. seems more consistent

Reply:
So ti's still in the code. 
https://github.com/rails/rails/blob/v7.1.3/activerecord/lib/active_record.rb#L246

So `respond_to` still returns true, then it raises an error.  I thought best not to try at all if the version was too high.